cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocProvisioner.java

```diff
@@ -503,4 +494,26 @@ String getRunKey(ProvisionerContext context) {
     }
     return CLUSTER_PREFIX + cleanedAppName + "-" + programRunInfo.getRun();
   }
+
+  private void validateAutoscalingFields(DataprocConf conf, Map<String, String> properties, boolean throwException) {
+    if (conf.isPredefinedAutoScaleEnabled()) {
+      //If predefined auto-scaling is enabled, then user should not send values for the following
+      if (properties.containsKey(DataprocConf.WORKER_NUM_NODES) ||
+        properties.containsKey(DataprocConf.SECONDARY_WORKER_NUM_NODES) ||
+        properties.containsKey(DataprocConf.AUTOSCALING_POLICY)) {
+
+        if (throwException) {
```

Comment:
Why do you want to have this code in the helper method? It's not reused as you have single call with throw=true and single with throw=false. It would be more clear if you have a helper method that just tells you `isAutoscalingValid` and then you can have a proper message in respective place. Or have three helper methods if you want more structure. boolean parameter hinders readability as when reading the code you don't see parameter name and don't know what does it affect

Reply:
thank you. addressed it.